pkg/infrastructure/vsphere/clusterapi/clusterapi.go

```diff
@@ -131,3 +135,12 @@ func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionIn
 
 	return nil
 }
+
+func downloadImage(installConfig *installconfig.InstallConfig) bool {
+	for _, fd := range installConfig.Config.VSphere.FailureDomains {
+		if fd.Topology.Template == "" {
```

Comment:
Wont this return true after the first iteration (if the statement is true). The PR comments suggest that each Failure Domain should be checked.